vllm/core/scheduler.py

```diff
@@ -1124,6 +1184,19 @@ def _schedule_chunked_prefill(self) -> SchedulerOutputs:
                                            curr_loras,
                                            enable_chunking=True)
 
+        prefilling = running_scheduled.prefill_seq_groups + prefills.seq_groups
+
+        prefilling = [
+            p for p in prefilling if self._will_still_be_prefilling(p)
+        ]
+
+        # Set slot counts for next iteration
+        self.prefill_slots_running = len(prefilling)
+        self.big_prefill_requests = len([
```

Comment:
I'm wondering if this block of code will handle any cases that are not covered by the increment we do within `_schedule_prefills` 

https://github.com/vllm-project/vllm/pull/10235/files#diff-0e64e07127ff20fc01be18cc95c0f58e0ff818e0dbabd5e8d2e07ca27943a7e4R1001-R1002

Reply:
Ah, yeah good catch. We should add a second iteration on the unit test for large requests to make sure that this code block counts the mid-prefill large request and the second one doesn't immediately start. Would you mind adding that?

Reply:
Done